### ISSUE TITLE
TypeError: Cannot read property 'catch' of undefined

### ISSUE THREAD

**Issue by stefanleoussis** [kibertoad/toad-scheduler#94]
### Hi there,
I've implemented the Async Task example used in the documentation. I am fairly confident I implemented this correctly, but I am receiving this output.

### Here is my code:
<img width="382" alt="Screen Shot 2022-05-15 at 11 15 13 PM" src="https://user-images.githubusercontent.com/58789967/168514382-35aca9c3-475c-4567-b02a-3d68ccbc9aa7.png">

### The Error:
<img width="930" alt="Screen Shot 2022-05-15 at 11 15 21 PM" src="https://user-images.githubusercontent.com/58789967/168514420-e814be55-0939-45bf-9cdd-2f3507d4f611.png">

### Lastly, the file to which the error is pointing to.
<img width="644" alt="Screen Shot 2022-05-15 at 11 16 16 PM" src="https://user-images.githubusercontent.com/58789967/168514456-92b8942b-8864-4bf1-bf06-f4ae7a0cef6d.png">

### Enviroment:
toad-scheduler Version: Both 1.6.0 and 1.0.2 have been tried with resulting in the same outputs
Node Version: 12.13.1
nodemon: ^1.19.1

Please let me know,
Thank you!

**Comment by AndreWillomitzer** [comment]
Hello, I contributed to the project a good while ago but I got an email notification about the repo. 

@kibertoad  hi buddy hope you've been well <3 I was wondering, do you think there could be some kind of default value to give to "handler" in case it's undefined? I notice there's a default set for errorHandler but not handler, and the error seems to say cannot read catch() of undefined on the "handler" part.  

As for why it's undefined, the only differences I see between your usage and the documentation is "seconds: 20" vs "seconds:1" and also "(err: Error)=>{}" notice the type def, versus "(err) =>{}". Also, I see that your scheduler.stop() has a setTimeout(). 

Did you try following the exact implementation of the docs?

**Comment by stefanleoussis** [comment]
@AndreWillomitzer 

Yes I have tried all following cases, except for defining the type, "(err: Error)" because I am writing this in js and I am unaware of how to create typedefs in js.

**Comment by kibertoad** [comment]
Based on the code that you have shared, `this.handler()` can return undefined in some cases. To handle such cases properly, you can do chaining:
```
const handlerPromise = this.handler()
return Promise.resolve(handlerPromise)
.catch() // etc
```

Alternatively, if you make handler `async`, its result should be wrapped into promise automatically, and your code should work out of the box.

**Comment by kibertoad** [comment]
@stefanleoussis Did that help you resolve your issue?

**Comment by kibertoad** [comment]
@stefanleoussis Sorry for not reaching out earlier. Actually, fix would have been as simple as returning the `promise1` from your code.

I definitely need to document this better.
